packages/cache/src/index.ts

```diff
@@ -0,0 +1,170 @@
+import { Redis } from 'ioredis';
+import { LRUCache } from 'lru-cache';
+import { logger } from '@prairielearn/logger';
+import * as Sentry from '@prairielearn/sentry';
+
+class Cache {
+  cacheEnabled = false;
+  cacheType = 'none';
+  memoryCache?: LRUCache<string, string>;
+  client?: Redis;
+  cacheKeyPrefix = 'prairielearn-cache:';
+
+  async init(config: {
+    cacheType: 'none' | 'memory' | 'redis';
+    cacheKeyPrefix: string;
```

Comment:
Bikeshedding: we already know these are the settings for a cache. Could we name these just `type` and `cachePrefix` instead?

Reply:
I am always in for reducing naming redundancies when possible. Updated to `type` and `keyPrefix` in [dacc892](https://github.com/PrairieLearn/PrairieLearn/pull/9360/commits/dacc892c32a53e09450e4b8ba921a9242e3bce7f).